onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -1963,6 +2003,14 @@ PYBIND11_MODULE(onnxruntime_pybind11_state, m) {
   addGlobalMethods(m, env);
   addObjectMethods(m, env);
 
+  Ort::SessionOptions tmp_options;
+  //the return status doesn't matter;
+  if(!OrtSessionOptionsAppendExecutionProvider_Dnnl(tmp_options, 0)){
+    const logging::Logger& default_logger = logging::LoggingManager::DefaultLogger();
+    LOGS(default_logger, INFO) << "This is a trick to initialize the shared provider bridge, it is safe to ignore.";
```

Comment:
Is there no other better way to initialize the shared provider bridge? This seems like a roundabout way to achieve it.

Reply:
agree. @RyanUnderhill , what do you think?

Reply:
(see above comment)

Reply:
Why do we need to expose a separate API to load the provider bridge? We're inside the core and should be able to invoke the appropriate functions directly.

Reply:
@pranavsharma , i think Ryan's point is currently the load provider bridge and load a provider shared library (like dnnl in my trick) is mixed in the same function, we just need to separate them. i will apply it to my pr first.